client/service/src/error.rs

```diff
@@ -48,6 +48,9 @@ pub enum Error {
 	#[error(transparent)]
 	Telemetry(#[from] sc_telemetry::Error),
 
+	#[error(transparent)]
+	Storage(#[from] sc_storage_monitor::Error),
```

Comment:
This should have been conditionally compiled based on optional `sc-storage-monitor`. Otherwise without Windows support, the whole `sc-service` no longer compiles on Windows.

P.S. I know there is https://github.com/paritytech/substrate/pull/13466 already open

Reply:
I forgot about windows. We definitely need a check in CI.

Reply:
No this was an oversight. This Error variant should not exist at all! @michalkucharczyk please remove it. For that we have the `Application` variant.

The entire idea around this storage monitor was that it is optional and downstream can either add it/not add it.